src/main/java/core/basesyntax/model/FruitTransaction.java

```diff
@@ -0,0 +1,73 @@
+package core.basesyntax.model;
+
+public class FruitTransaction {
+    private Operation operation;
+    private String fruit;
+    private int quantity;
+
+    public FruitTransaction(Operation operation, String fruit, int quantity) {
+        this.operation = operation;
+        this.fruit = fruit;
+        this.quantity = quantity;
+    }
+
+    public Operation getOperation() {
+        return operation;
+    }
+
+    public void setOperation(Operation operation) {
+        this.operation = operation;
+    }
+
+    public String getFruit() {
+        return fruit;
+    }
+
+    public void setFruit(String fruit) {
+        this.fruit = fruit;
+    }
+
+    public int getQuantity() {
+        return quantity;
+    }
+
+    public void setQuantity(int quantity) {
+        this.quantity = quantity;
+    }
+
+    @Override
+    public String toString() {
+        return "FruitTransaction{"
+                + "operation=" + operation
+                + ", fruit='" + fruit
+                + '\'' + ", quantity=" + quantity
+                + '}';
+    }
+
+    public enum Operation {
+        BALANCE("b"),
+        SUPPLY("s"),
+        PURCHASE("p"),
+        RETURN("r");
+
+        private String code;
+
+        Operation(String code) {
+            this.code = code;
+        }
+
+        public String getCode() {
+            return code;
+        }
+
+        public static Operation fromCode(String code) {
+            for (Operation op : values()) {
+                if (op.getCode().equalsIgnoreCase(code)) {
+                    return op;
+                }
+            }
+            throw new IllegalArgumentException("Invalid code: " + code);
+        }
+    }
+}
```

Comment:
The class `FruitTransaction` is well-defined and encapsulated. However, according to the checklist, we should not start the class implementation with an empty line. Please remove the empty line at the start of the class definition (line 2).